### PR TITLE
feat(framework-core): main interface for create review on a pull request

### PR DESCRIPTION
**Pull request by TomKristie**
Main interface for frame-work core, documentation + review comments:
1. renamin `comment` to `reviewPullRequest`
2. more tests!

PR commit for #105

### CODE REVIEW

File: src/default-options-handler.ts [hunk 1]
```diff
@@ -46,4 +52,23 @@ function addPullRequestDefaults(
   return pullRequestSettings;
 }
 
-export {addPullRequestDefaults};
+/**
+ * Format user input for pull request review comments
+ * @param options The user's options input for review comments
+ * @returns the formatted version of user input for pull request review comments
+ */
+export function addRevuewCommentsDefaults(
```
Comment (chingor13) [inline src/default-options-handler.ts:60]:
```suggestion
export function addReviewCommentsDefaults(
```

File: src/index.ts [hunk 2]
```diff
@@ -20,14 +20,71 @@ import {
...
+    );
+    return;
+  }
+  const gitHubConfigs = addRevuewCommentsDefaults(options);
+  const remote: RepoDomain = {
+    owner: gitHubConfigs.owner,
+    repo: gitHubConfigs.repo,
+  };
+  logger.info(
+    `Successfully created a review on pull requestt: ${gitHubConfigs.pullNumber}.`
```
Comment (chingor13) [inline src/index.ts:77]:
```suggestion
    `Successfully created a review on pull request: ${gitHubConfigs.pullNumber}.`
```

File: src/default-options-handler.ts [hunk 3]
```diff
@@ -46,4 +52,23 @@ function addPullRequestDefaults(
...
+    repo: options.repo,
+    owner: options.owner,
+    pullNumber: options.pullNumber,
+    // if zero set as 0
+    pageSize:
+      options.pageSize === null || options.pageSize === undefined
+        ? DEFAULT_PAGE_SIZE
+        : options.pageSize,
+  };
+  return reviewCommentSettings;
```
Comment (chingor13) [inline src/default-options-handler.ts:73]: Let's declare the return type in the signature and then we can return the object directly.

File: src/index.ts [hunk 4]
```diff
@@ -20,14 +20,71 @@ import {
...
+    rawChanges === null ||
+    rawChanges === undefined ||
+    rawChanges.size === 0
+  ) {
+    logger.info(
+      'Empty changes provided. No suggestions to be made. Cancelling workflow.'
+    );
+    return;
+  }
+  const gitHubConfigs = addRevuewCommentsDefaults(options);
```
Comment (chingor13) [inline src/index.ts:71]: Can we name this `createReviewComment` as it's a `CreateReviewComment` type?

**Review by chingor13 — CHANGES_REQUESTED** [review]
Thanks Kristie!

**Review by bcoe — APPROVED** [review]
This is looking solid to me once @chingor13's typo and naming nits are addressed.

**Review by chingor13 — APPROVED** [review]

**Pull request merged**